examples/proposed/lock.w

```diff
@@ -0,0 +1,83 @@
+bring cloud;
+
+/**
+ * A lock that can only be acquired by one inflight at a time.
+ *
+ * Internally, this uses a cloud.Counter to keep track of a number.
+ * If the number is 0, the lock is free. If the number is 1, the lock
+ * is acquired.
+ *
+ * This is a very simple lock, and it has a number of limitations:
+  * - It is not reentrant. If a single inflight tries to acquire the lock
+  *   twice, it may block forever.
+  * - It is not fair. If multiple inflights are waiting to acquire the lock,
+  *   there is no guarantee that they will acquire the lock in the order
+  *   that they requested it.
+  * - It is not stable. If an inflight crashes while holding the lock, the
+  *   lock will be permanently stuck in the acquired state.
+ */
+resource Lock {
+  counter: cloud.Counter;
```

Comment:
I feel like Counter is not a sufficient structure here.

Using something that supports:
- key/value storage
- TTLs

Would allow the lock to be both reentrant and "stable" in crash conditions.

Obviously we don't have that now, but the closest thing could be some kind of `Cache` I guess

Reply:
DynamoDB has support for TTLs so that might be something to consider.

Reply:
Yeah I think under-the-hood dynamo would be okay for this, although a little slow compared to Elasticache if you're okay with a volatile lock.